File: frontEnd/app/model/question.ts
export class Question {
    private questionText: String;
    private correctAnswers: Array<String>;
    private badAnswers: Array<String>;
    private categories: Array<String>;

    constructor(questionText: String, correctAnswers: Array<String>, badAnswers: Array<String>, categories: Array<String>) {
        this.questionText = questionText;
        this.correctAnswers = correctAnswers;
        this.badAnswers = badAnswers;
        this.categories = categories;
    }
    
    getBadAnswers(): Array<String> {
        return this.badAnswers;
    }

}

File: backEnd/src/main/java/pl/iqa/model/question/Question.java
package pl.iqa.model.question;

import pl.iqa.model.question.validation.AtLeastOneCorrect;

import javax.validation.Valid;
import javax.validation.constraints.NotNull;
import java.util.List;
import java.util.Objects;

public class Question {

    @NotNull
    private String questionText;

    @AtLeastOneCorrect
    @Valid
    private List<Answer> answers;

    @Valid
    private List<Category> categories;

    public String getQuestionText() {
        return questionText;
    }

    public void setQuestionText(String questionText) {
        this.questionText = questionText;
    }

    public List<Answer> getAnswers() {
        return answers;
    }

    public void setAnswers(List<Answer> answers) {
        this.answers = answers;
    }

    public List<Category> getCategories() {
        return categories;
    }

    public void setCategories(List<Category> categories) {
        this.categories = categories;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        Question question = (Question) o;
        return Objects.equals(questionText, question.questionText) &&
                Objects.equals(answers, question.answers) &&
                Objects.equals(categories, question.categories);
    }

    @Override
    public int hashCode() {
        return Objects.hash(questionText, answers, categories);
    }
}
